Added implementation for managed shares

allenktv opened the pull request:
In conjunction with PR: #78

illiminable commented:
Landing the functionality for now, but we need to follow up on

1) UI review for icons
2) User feedback during load time.

allenktv commented:
@illiminable I implemented `.loading` to have a spinner loading when we are currently waiting for shares, however I found that, while it is true, it fully replaces the dropdown portion, and we can't see anything on  the list. 

Currently, I have this implemented so that, when it is loaded, it will show the loading spinner until one of the following happens:

1. A managed share is found.
2. We get the results for the found network shares, whether empty or not.

So this means that the spinner will disappear if a managed share is found, even though we are still waiting on network shares. I know this is different than what we talked about, but let me know how this sounds to you.

In the meanwhile, I will try to dig in a little bit in the code and see if it is possible for it to do what we want.

allenktv commented:
I added removed the icon for found network shares. I also changed when the loading spinner turns off. Now it only turns off when we get the results for network shares (empty or not)
